File: project-euler/80.py

from decimal import *
getcontext().prec = 200

def main():
    total = 0
    for n in range(2, 101):
        s = Decimal(n).sqrt()
        if round(s) != s:
            for i in range(100):
                total += int(s % 10)
                s *= 10
    return total

if __name__ == '__main__':
    print(main())

File: project-euler/80.jl


function solve()
    total = 0
    for n in 1:100
        s = setprecision(5000) do
            sqrt(BigFloat(n))
        end
        if s != round(s)
            println(">>> ", n)
            for i in 1:100
                s *= 10
                println(i, ' ', convert(Int, round(s % 10)))
                total += convert(Int, round(s % 10))
            end
            return total
        end
    end
end

println(solve())
